core/src/package.rs

```diff
@@ -10,19 +13,21 @@ use crate::package::PackageImplementation::Native;
 use crate::{error::CoreError, OutputFormat};
 
 /// Transform from a node into another node
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct Transform {
     pub from: String,
     pub to: Vec<OutputFormat>,
     pub description: Option<String>,
     pub arguments: Vec<ArgInfo>,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct ArgInfo {
     pub name: String,
-    pub default: Option<String>,
+    pub default: Option<Value>,
     pub description: String,
+    #[serde(default = "default_arg_type")]
+    pub r#type: ArgType,
```

Comment:
I think this is fine since the actual argument name is `type`, and will be serialised and deserialised as `type`. We could change it to `arg_type`, but that is less clean and also longer. I think it makes more sense to not have `arg_`-prefixes for fields within an `ArgInfo`, since the context gives that it is indeed related to an argument (we have `name`, not `arg_name`, `description`, not `arg_description` etc).

But sure, I can change this if you think that's important! But do you agree to `serde(rename)`-ing it to `type`?